requery/src/main/java/io/requery/sql/RawEntityNamedQuery.java

```diff
@@ -0,0 +1,103 @@
+package io.requery.sql;
+
+import io.requery.PersistenceException;
+import io.requery.meta.Attribute;
+import io.requery.meta.Type;
+import io.requery.query.BaseResult;
+import io.requery.query.Result;
+import io.requery.util.CloseableIterator;
+import io.requery.util.function.Predicate;
+import io.requery.util.function.Supplier;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Raw Query with Named parameters
+ *
+ * @author debop (Sunghyouk Bae)
+ * @since 18. 12. 8
+ */
+public class RawEntityNamedQuery<E extends S, S> extends PreparedQueryOperation implements Supplier<Result<E>> {
+
+    private final EntityReader<E, S> reader;
+    private final Type<E> type;
+    private final BoundParameters boundParameters;
+    private final String sql;
+
+    RawEntityNamedQuery(EntityContext<S> context, Class<E> cls, String sql, Map<String, Object> namedParameters) {
+        super(context, null);
+
+        NamedParameterParser inlined = new NamedParameterParser(sql, namedParameters).apply();
+        this.type = configuration.getModel().typeOf(cls);
+        this.sql = inlined.sql();
+        this.reader = context.read(cls);
+        this.boundParameters = new BoundParameters(inlined.parameters());
+    }
+
+    @Override
+    public Result<E> get() {
+        PreparedStatement statement = null;
+        try {
+            Connection connection = configuration.getConnection();
+            statement = prepare(sql, connection);
+            mapParameters(statement, boundParameters);
+            return new EntityResult(statement);
+        } catch (Exception e) {
+            throw StatementExecutionException.closing(statement, e, sql);
+        }
+    }
+
+    private class EntityResult extends BaseResult<E> {
```

Comment:
This looks identical to `RawEntityQuery.EntityResult` this class should probably extend RawEntityQuery to avoid duplication

Reply:
Can I change RawEntityQuery overridable? If then, I will